grpclb/src/main/java/io/grpc/grpclb/GrpclbLoadBalancer.java

```diff
@@ -249,6 +255,7 @@ public void transportShutdown(
     handleError(status.augmentDescription("Transport to LB server closed"));
     synchronized (lock) {
       if (transport == lbTransport) {
+        directTransport = null;
```

Comment:
This seems fine, but you may consider moving this line to `connectToLb()` (possibly above `closed` check) to keep reset logic all in the same place. It's fine whichever place you prefer.